osu.Game.Rulesets.Osu/Difficulty/Preprocessing/OsuDifficultyHitObject.cs

```diff
@@ -89,17 +91,76 @@ private void setDistances(double clockRate)
             if (lastObject is Slider lastSlider)
             {
                 computeSliderCursorPosition(lastSlider);
-                TravelDistance = lastSlider.LazyTravelDistance * scalingFactor;
+                TravelDistance = 0;
                 TravelTime = Math.Max(lastSlider.LazyTravelTime / clockRate, min_delta_time);
                 MovementTime = Math.Max(StrainTime - TravelTime, min_delta_time);
+                MovementDistance = Vector2.Subtract(lastSlider.TailCircle.StackedPosition, BaseObject.StackedPosition).Length * scalingFactor;
+
+                int repeatCount = 0;
+
+                Vector2 currSliderPosition = ((OsuHitObject)lastSlider.NestedHitObjects[0]).StackedPosition;
+
+                for (int i = 1; i < lastSlider.NestedHitObjects.Count; i++)
```

Comment:
There needs to be commenting on what this entire loop is doing, because I can't understand it. It seems to basically be doing another lazy approximation of travel distance, but I don't know how it differs from `LazyTravelDistance` without running through the calculations manually.

Reply:
This entire section has been moved and edited to be ran inside of the lazy slider distance calculation, since you're right in that the loop is basically the same but with some adjustments to better approximate real movement patterns.